src/syntax/binding_when_syntax.ts

```diff
@@ -39,6 +39,15 @@ class BindingWhenSyntax<T> implements interfaces.BindingWhenSyntax<T> {
         return new BindingOnSyntax<T>(this._binding);
     }
 
+    public whenTargetMultiTagged(...tags: [interfaces.Tag, ...interfaces.Tag[]]): interfaces.BindingOnSyntax<T> {
+        this._binding.constraint = (request: interfaces.Request) =>
```

Comment:
There is an issue with the current typings where the Binding class and the Binding interface differ.

Binding interface
```
constraint: ConstraintFunction;

export interface ConstraintFunction extends Function {
        metaData?: Metadata;
        (request: Request | null): boolean;
}
```

vs - public constraint: (request: interfaces.Request) => boolean;

Although I do not think we call with null it can be through the interface so you need to check for it

You can see the taggedConstraint helper does

```
const taggedConstraint = (key: string | number | symbol) => (value: any) => {

    const constraint: interfaces.ConstraintFunction =  (request: interfaces.Request | null) =>
        request !== null && request.target !== null && request.target.matchesTag(key)(value);

    constraint.metaData = new Metadata(key, value);

    return constraint;
};
```

Reply:
This would be a breaking change but should ConstraintFunction change to have metadata property as a Metadata array ?

Reply:
But we could add a multiMetaData property and look for that in https://github.com/inversify/InversifyJS/blob/4332cb8746327cc70b8fa48016cf33786b11909a/src/utils/serialization.ts#L44

Reply:
When you change to `(request: interfaces.Request | null)` there needs to be full coverage

Reply:
> This would be a breaking change but should ConstraintFunction change to have metadata property as a Metadata array ?

I see, I would potentially be a breaking change if we change that because we expose the `taggedConstraint` with no type defined. 

Although the `metaData` here is only used for the message logging and we could potentially change that. This is something that we only expect users to write and not read. So making it an `Metadata | Metadata[]` is something that is not as harmful.

Reply:
> So making it an Metadata | Metadata[] is something that is not as harmful.

Still a breaking change.

It's not that important but for consistenct error messages for bindings with tagged constraints...

When we do v6 we can change to
```
export interface ConstraintFunction extends Function {
        metaData?: Metadata[];
        (request: Request | null): boolean;
}
```

Reply:
It makes sense. So for now we can keep the metadata null when there is multiple tags on v5 and provide a better error message on v6.

Reply:
I would do multiMetadata now personally but I will not block the pr if it is omitted

Reply:
And introduce the breaking change in v5? Or do we create a new property into `ConstraintFunction`?

Reply:
Add optional multiMetadata property.